crates/turbo-tasks/src/graph/with_future.rs

```diff
@@ -29,18 +32,23 @@ impl<T, H> Future for With<T, H>
 where
     T: Future,
 {
-    type Output = (H, T::Output);
+    type Output = (H, Span, T::Output);
 
     fn poll(
         self: std::pin::Pin<&mut Self>,
         cx: &mut std::task::Context<'_>,
     ) -> std::task::Poll<Self::Output> {
         let this = self.project();
+        let guard = this.span.enter();
         match this.future.poll(cx) {
-            std::task::Poll::Ready(result) => std::task::Poll::Ready((
-                this.handle.take().expect("polled after completion"),
-                result,
-            )),
+            std::task::Poll::Ready(result) => {
+                drop(guard);
```

Comment:
Isn't that equivalent to `instrument`ing the future directly? Why do we need to keep the span as part of this struct instead?